frame/staking/src/lib.rs

```diff
@@ -864,6 +1023,79 @@ decl_module! {
 			}
 		}
 
+		/// Submit a phragmen result to the chain. If the solution:
+		/// 1. is valid
+		/// 2. has a better score than a potentially existing solution on chain
+		///
+		/// it will replace it.
+		///
+		/// A solution consists of two pieces of data:
+		/// 1. `winners`: a flat vector of all the winners of the round.
+		/// 2. `assignments`: the compact version of an assignment vector that encodes the edge
+		///    weights.
+		///
+		/// Both of which may be computed using the [`phragmen`], or any other algorithm.
+		///
+		/// A solution is valid if
+		/// 1. All the presented winners are actually on-chain candidates.
+		/// 2. All the presented edges contain:
+		///   - a voter who is an on-chain nominator.
+		///   - a target who is an on-chain candidate.
+		///   - that target must actually be among the nominations of the voter.
+		///
+		/// A solutions score is consisted of 3 parameters:
+		/// 1. `min { support.total }` for each support of a winner. This value should be maximized.
+		/// 2. `sum { support.total }` for each support of a winner. This value should be minimized.
+		/// 3. `sum { support.total^2 }` for each support of a winner. This value should be
+		///    minimized (to ensure less variance)
+		///
+		/// # <weight>
+		/// major steps:
+		/// - decode from compact: E(E)
+		/// - build_support_map: O(E)
+		/// - evaluate_support: E(E)
+		/// - 1 read which decodes a `ElectionResult` from storage.
+		/// - 1 read which decodes a `ElectionStatus` from storage.
+		/// - `2*W` reads each of which decodes an `AccountId` and a `ValidatorPrefs` to ensure
+		///   winner veracity.
+		/// - N reads from `Bonded`, `Validators` and `Nominators` to ensure nominator veracity. N
+		///   is the number of nominators.
+		/// - O(N * E') to ensure nomination veracity. (E' is the average voter count per nominator
+		///   and it is less than `MAX_NOMINATIONS`)
+		/// # </weight>
+		#[weight = SimpleDispatchInfo::FixedNormal(10_000_000)]
+		fn submit_election_solution(
+			origin,
+			winners: Vec<T::AccountId>,
+			compact_assignments: CompactAssignments<T::AccountId, ExtendedBalance>,
+		) {
+			let _who = ensure_signed(origin)?;
+			Self::check_and_replace_solution(
+				winners,
+				compact_assignments,
+				ElectionCompute::Signed
+			)?
+		}
+
+		/// Unsigned version of `submit_election_solution`. Will only be accepted from those who are
+		/// in the current validator set.
+		fn submit_election_solution_unsigned(
+			origin,
+			winners: Vec<T::AccountId>,
+			compact_assignments: CompactAssignments<T::AccountId, ExtendedBalance>,
+			// already used and checked.
+			_validator_index: u32,
+			_signature: <T::KeyType as RuntimeAppPublic>::Signature,
+		) {
+			ensure_none(origin)?;
+			Encode::encode(&_signature);
```

Comment:
What's the purpose of this? Since we use the variable we should drop `_` prefix.

Reply:
Me forgetting to remove stuff while debugging :D